File: Cpp/Largest Number.cpp
// 4ms, 99.47%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	string largestNumber(const vector<int> &nums)
	{
		vector<string> tmp;
		tmp.reserve(nums.size());
		for(const auto n : nums) tmp.push_back(to_string(n));
		sort(tmp.begin(), tmp.end(), [](auto &x, auto &y) {return x + y > y + x; });
		string ret = accumulate(tmp.begin(), tmp.end(), string());
		return all_of(ret.begin(), ret.end(), [](auto x) {return x == '0'; }) ? "0" : ret;
	}
};

File: Python3/Largest Number.py
# 64ms, 99.18%
class Solution:
    def largestNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        from functools import cmp_to_key
        comp = lambda a, b: 1 if a + b < b + a else -1 if a + b > b + a else 0
        return str(int(''.join(sorted([str(x) for x in nums], key=cmp_to_key(comp)))))
